### DISCUSSION TITLE
Upgrading from v2 to v3

### DISCUSSION THREAD

**Discussion by ghost** [operator-framework/java-operator-sdk#1351] [Q&A]
Is there any guide what changes I need to make to my operator when upgrading from v2 to v3? I tried simply bumping the version, and I'm running into API changes that are not obvious to resolve.

Specifically, both implementations of ErrorStatusHandler and EventSourceInitializer in my Reconciler class need to be updated, but I'm not finding docs or examples of what exactly needs to happen there now.

**Comment by csviri** [comment]
Hi @stefanbethke , there is a migration guide:
https://javaoperatorsdk.io/docs/v3-migration

Although not specifies the details rather just mentions what parts are changed. The up to date docs explains the current interface.
Also the samples and Integration tests should serve to explain the details.

`ErrorStatusHandler` has just a minimal update, it contains now the context not just a retry info (that is accessible from context), and handles not just RuntimeExceptions but also checked Exceptions.

`EventSourceInitializer` now is just using named event sources, it is straight forward to migrate this, see usage of: 

`EventSourceInitializer.nameEventSources` here:

https://github.com/java-operator-sdk/java-operator-sdk/blob/0d57629830cdd9ac1c3665438a1493ca438afdc4/sample-operators/webpage/src/main/java/io/javaoperatorsdk/operator/sample/WebPageReconciler.java#L78-L78
